terminus-ui/selection-list/src/selection-list.component.ts

```diff
@@ -561,12 +561,10 @@ export class TsSelectionListComponent implements
           .subscribe(newValue => {
             // istanbul ignore else
             if (newValue) {
-              if (!newValue[0]) {
-                return;
+              if (newValue[0]) {
+                this.searchQuery = this.displayFormatter(newValue[0]);
```

Comment:
With this in place, the value will get set twice. Once here and once on line 571. Seems like we probably just want to remove this internal `if` on line 564?

Reply:
Here is only to set the `searchQuery` instead of form control value, and line 571 sets the form control value, right?
We cannot remove the `if` statement, because when it's an empty array, you can't set format on that. This change is just to let empty array go through. Previously, if new value is an empty array, the stream stops, so `selectionListFormControl` value never gets set properly.

Reply:
Maybe I'm unclear on the extent of the bug you are solving? The issue calls out the value of the form control is not updated correctly. I get why removing the return was needed to solve that, but I'm unclear why the searchQuery was edited?

Reply:
The searchQuery needs to be there for all the new value except when it's an empty array. Previously, because when it's an empty array, the stream returns, so no `if` condition needed for setting searchQuery. Now we don't want to stop the stream just because it's an empty array, thus this `if` is needed because there is no point to set formatter on an empty array. 
The issue is about setting an empty array on a form control.